axelrod/actions.py

```diff
@@ -9,31 +9,65 @@
 C, D = Actions.C, Actions.D
 """
 
-# Type alias for actions.
-Action = str
+from enum import Enum
+from typing import Iterable
+
+
+class UnknownActionError(ValueError):
+    def __init__(self, *args):
+        super(UnknownActionError, self).__init__(*args)
+
+
+class Actions(Enum):
+
+    C = 1
+    D = 0
+
+    def __bool__(self):
+        return bool(self.value)
 
+    def __repr__(self):
+        return '{}'.format(self.name)
 
-class Actions(object):
-    C = 'C'  # type: Action
-    D = 'D'  # type: Action
+    def __str__(self):
+        return '{}'.format(self.name)
+
+    def flip(self):
+        """Returns the opposite Action. """
+        if self == Actions.C:
+            return Actions.D
+        if self == Actions.D:
+            return Actions.C
+
+    @classmethod
+    def from_char(cls, character):
+        """Converts a single character into an Action. `Action.from_char('C')`
+        returns `Action.C`. `Action.from_char('CC')` raises an error. Use
+        `str_to_actions` instead."""
+        if character == 'C':
+            return cls.C
+        elif character == 'D':
+            return cls.D
+        else:
+            raise UnknownActionError('Character must be "C" or "D".')
```

Comment:
That's how it was before I think, I suggested this way thinking it would be more efficient.